src/Uno.Toolkit.UI/Extensions/LoadableExtension.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Uno.Disposables;
+
+namespace Uno.Toolkit.UI
+{
+	internal static class LoadableExtension
+	{
+		public static IDisposable BindIsExecuting(this ILoadable source, Action update, bool propagateInitialValue = true) => source.BindIsExecuting(_ => update(), propagateInitialValue);
+
+		public static IDisposable BindIsExecuting(this ILoadable source, Action<bool> update, bool propagateInitialValue = true)
+		{
+			if (propagateInitialValue) update(source.IsExecuting);
+
+			source.IsExecutingChanged += OnIsExecutingChanged;
```

Comment:
What's the expected thread affinity here? What happens if it is raised from a background thread?

Reply:
made the changes in the caller, so the `update` here will check for thread-access to invoke directly or from dispatcher